<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1315</width>
    <height>1121</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_6">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_26">
      <item>
       <widget class="QPushButton" name="scene01_pushButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Scene01: room</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="scene02_pushButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Scene02: cup</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="scene03_pushButton">
        <property name="text">
         <string>Scene03:  VeachMIS</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="myscene_pushButton">
        <property name="text">
         <string>My Scene: Iron Man</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="yourscene_pushButton">
        <property name="text">
         <string>Your Scene</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_25">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QGroupBox" name="camera_groupBox">
          <property name="title">
           <string>Camera Set</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="0" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout_21">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout">
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_6">
                 <item>
                  <widget class="QLabel" name="label_4">
                   <property name="text">
                    <string>Position:</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="camera_lineEdit_position_x">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="camera_lineEdit_position_y">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="camera_lineEdit_position_z">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_7">
                 <item>
                  <widget class="QLabel" name="label_5">
                   <property name="text">
                    <string>Look at:</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="camera_lineEdit_lookat_x">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="camera_lineEdit_lookat_y">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="camera_lineEdit_lookat_z">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_8">
                 <item>
                  <widget class="QLabel" name="label_6">
                   <property name="text">
                    <string>Up:</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="camera_lineEdit_up_x">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="camera_lineEdit_up_y">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="camera_lineEdit_up_z">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_9">
                 <item>
                  <widget class="QLabel" name="label_7">
                   <property name="text">
                    <string>FovY:</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="camera_lineEdit_fovy">
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_8">
                   <property name="text">
                    <string>Â°</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_12">
                 <item>
                  <layout class="QHBoxLayout" name="horizontalLayout_10">
                   <item>
                    <widget class="QLabel" name="label_9">
                     <property name="text">
                      <string>Width:</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLineEdit" name="camera_lineEdit_width">
                     <property name="text">
                      <string/>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <layout class="QHBoxLayout" name="horizontalLayout_11">
                   <item>
                    <widget class="QLabel" name="label_10">
                     <property name="text">
                      <string>Height:</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLineEdit" name="camera_lineEdit_height">
                     <property name="text">
                      <string/>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPushButton" name="camera_pushButton_apply">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Apply</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="lightSoure_groupBox">
          <property name="title">
           <string>Light Source Set</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="1" column="0">
            <widget class="QGroupBox" name="sphereLight_groupBox">
             <property name="title">
              <string>Sphere Light</string>
             </property>
             <property name="checkable">
              <bool>false</bool>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
             <layout class="QGridLayout" name="gridLayout">
              <item row="0" column="0">
               <layout class="QHBoxLayout" name="horizontalLayout_23">
                <item>
                 <layout class="QVBoxLayout" name="verticalLayout_3">
                  <item>
                   <layout class="QHBoxLayout" name="horizontalLayout_13">
                    <item>
                     <widget class="QLabel" name="label_13">
                      <property name="text">
                       <string>Position:</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="sphereLight_lineEdit_position_x"/>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="sphereLight_lineEdit_position_y"/>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="sphereLight_lineEdit_position_z"/>
                    </item>
                   </layout>
                  </item>
                  <item>
                   <layout class="QHBoxLayout" name="horizontalLayout_14">
                    <item>
                     <widget class="QLabel" name="label_14">
                      <property name="text">
                       <string>Radius:</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="sphereLight_lineEdit_radius"/>
                    </item>
                   </layout>
                  </item>
                  <item>
                   <layout class="QHBoxLayout" name="horizontalLayout_15">
                    <item>
                     <widget class="QLabel" name="label_12">
                      <property name="text">
                       <string>Le:</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="sphereLight_lineEdit_le_x"/>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="sphereLight_lineEdit_le_y"/>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="sphereLight_lineEdit_le_z"/>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </item>
                <item>
                 <widget class="QPushButton" name="sphereLight_pushButton_add">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>Add</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QGroupBox" name="quadLight_groupBox">
             <property name="title">
              <string>Quad Light</string>
             </property>
             <property name="checkable">
              <bool>false</bool>
             </property>
             <layout class="QGridLayout" name="gridLayout_3">
              <item row="0" column="0">
               <layout class="QHBoxLayout" name="horizontalLayout_22">
                <item>
                 <layout class="QVBoxLayout" name="verticalLayout_2">
                  <item>
                   <layout class="QHBoxLayout" name="horizontalLayout_5">
                    <item>
                     <widget class="QLabel" name="label_15">
                      <property name="text">
                       <string>Center:</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="quadLight_lineEdit_center_x"/>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="quadLight_lineEdit_center_y"/>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="quadLight_lineEdit_center_z"/>
                    </item>
                   </layout>
                  </item>
                  <item>
                   <layout class="QHBoxLayout" name="horizontalLayout_4">
                    <item>
                     <widget class="QLabel" name="label_19">
                      <property name="text">
                       <string>Size:</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="quadLight_lineEdit_size_w"/>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="quadLight_lineEdit_size_h"/>
                    </item>
                   </layout>
                  </item>
                  <item>
                   <layout class="QHBoxLayout" name="horizontalLayout_3">
                    <item>
                     <widget class="QLabel" name="label_17">
                      <property name="text">
                       <string>Normal:</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="quadLight_lineEdit_normal_x"/>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="quadLight_lineEdit_normal_y"/>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="quadLight_lineEdit_normal_z"/>
                    </item>
                   </layout>
                  </item>
                  <item>
                   <layout class="QHBoxLayout" name="horizontalLayout_2">
                    <item>
                     <widget class="QLabel" name="label_18">
                      <property name="text">
                       <string>Up:</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="quadLight_lineEdit_up_x"/>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="quadLight_lineEdit_up_y"/>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="quadLight_lineEdit_up_z"/>
                    </item>
                   </layout>
                  </item>
                  <item>
                   <layout class="QHBoxLayout" name="horizontalLayout">
                    <item>
                     <widget class="QLabel" name="label_16">
                      <property name="text">
                       <string>Le:</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="quadLight_lineEdit_le_x"/>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="quadLight_lineEdit_le_y"/>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="quadLight_lineEdit_le_z"/>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </item>
                <item>
                 <widget class="QPushButton" name="quadLight_pushButton_add">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>Add</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="pathTracer_groupBox">
          <property name="title">
           <string>Path Tracer Set</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout_24">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_4">
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_16">
                 <item>
                  <widget class="QLabel" name="label">
                   <property name="text">
                    <string>Samples Per Pixelï¼</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="pathTracer_lineEdit_sperpixel"/>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_17">
                 <item>
                  <widget class="QLabel" name="label_2">
                   <property name="text">
                    <string>Samples Per Light Sourceï¼</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="pathTracer_lineEdit_sperlight"/>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_18">
                 <item>
                  <widget class="QLabel" name="label_3">
                   <property name="text">
                    <string>Maximum Depth:</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="pathTracer_lineEdit_maxdepth"/>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_19">
                 <item>
                  <widget class="QLabel" name="label_11">
                   <property name="text">
                    <string>Ambient lightï¼</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="pathTracer_lineEdit_ambient_x"/>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="pathTracer_lineEdit_ambient_y"/>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="pathTracer_lineEdit_ambient_z"/>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPushButton" name="pathTracer_pushButton_apply">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Apply</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="Display" name="display" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>512</width>
            <height>512</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_20">
          <item>
           <widget class="QPushButton" name="render_pushButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Render</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextBrowser" name="textBrowser">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1315</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuLoad_Scene">
    <property name="title">
     <string>Load Scene</string>
    </property>
    <addaction name="actionScene01_room"/>
    <addaction name="actionScene02_cup"/>
    <addaction name="actionScene03_VeachMIS"/>
    <addaction name="actionMy_Scene_Iron_Man"/>
   </widget>
   <widget class="QMenu" name="menuSave_Image">
    <property name="title">
     <string>Save Image</string>
    </property>
    <addaction name="actionSave_as"/>
   </widget>
   <widget class="QMenu" name="menuOpen_OBJ">
    <property name="title">
     <string>Open OBJ</string>
    </property>
    <addaction name="actionLoad"/>
   </widget>
   <addaction name="menuLoad_Scene"/>
   <addaction name="menuOpen_OBJ"/>
   <addaction name="menuSave_Image"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="separator"/>
   <addaction name="actionScene01_room"/>
   <addaction name="actionScene02_cup"/>
   <addaction name="actionScene03_VeachMIS"/>
   <addaction name="actionMy_Scene_Iron_Man"/>
   <addaction name="actionLoad"/>
   <addaction name="actionSave_as"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionScene01_room">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icon/room.png</normaloff>:/icon/room.png</iconset>
   </property>
   <property name="text">
    <string>Scene01: room</string>
   </property>
  </action>
  <action name="actionScene02_cup">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icon/cup.png</normaloff>:/icon/cup.png</iconset>
   </property>
   <property name="text">
    <string>Scene02: cup</string>
   </property>
  </action>
  <action name="actionScene03_VeachMIS">
   <property name="icon">
    <iconset>
     <normaloff>icon/VeachMIS.png</normaloff>
     <normalon>:/icon/VeachMIS.png</normalon>icon/VeachMIS.png</iconset>
   </property>
   <property name="text">
    <string>Scene03: VeachMIS</string>
   </property>
  </action>
  <action name="actionMy_Scene_Iron_Man">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icon/Iron Man.png</normaloff>:/icon/Iron Man.png</iconset>
   </property>
   <property name="text">
    <string>My Scene: Iron Man</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icon/open.png</normaloff>:/icon/open.png</iconset>
   </property>
   <property name="text">
    <string>Load *.obj</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="icon">
    <iconset resource="icon.qrc">
     <normaloff>:/icon/save.png</normaloff>:/icon/save.png</iconset>
   </property>
   <property name="text">
    <string>Save as *.png</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Display</class>
   <extends>QWidget</extends>
   <header>Display.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icon.qrc"/>
 </resources>
 <connections/>
</ui>
